rails-new.md

```diff
@@ -1,77 +1,129 @@
-Finally, Rails runs the bundle install command to install the dependencies listed in your Gemfile .
-Once everything is set up, navigate to the sharkapp directory:
-You can now start the Rails server to ensure that your application is working, using the rails server command. If you are working on your local machine, type:
-Rails binds to localhost by default, so you can now access your application by navigating your browser to locahost:3000, where you will see the following image:
-             lite3
-            cd sharkapp
-    rails server
-  
-   If you are working on a development server, first ensure that connections are allowed on port 3000 :
-Then start the server with the --binding flag, to bind to your server IP:
-Navigate to http://your_server_ip:3000 in your browser, where you will see the Rails welcome message.
-Once you have looked around, you can stop the server with CTRL+C .
-   sudo ufw allow 3000
-   rails server --binding=your_server_ip
-   Rails Landing Page
-  
- With your application created and in place, you are ready to start building from the Rails boilerplate to create a unique application.
+# Step 1 — Creating an Empty Container
+
+We Are going to house our Rails Application within a container, so in order to do this, we must first create it! On your local machine, make an empty directory.
+
+```
+mkdir bnb-library
+cd bnb-library
+```
+
+Instead of having to manage the versions of multiple different dependancies for your application such as *Ruby*, *Rails*, and *Postgres*, we maintain our own set of images that are:
+- Based on Ruby
+- Have PostgreSQL installed (you might opt for mysql-client)
+- Have NodeJS installed (for compiling assets)
+- There is some configuration of bundler so that it will install our gems in the `/bundle` directory.
+
+In your new directory make a `docker-compose.yml` file and paste in:
+
+```
+version: "3"
+services:
+  app:    
+    image: brandnewbox/bnb-ruby:2.7-postgresql
+    command: bundle exec puma -C config/puma.rb
+    env_file: .docker-development-vars
+    volumes:
+      - .:/app:cached
+      - bundle_cache:/usr/local/bundle
+    ports:
+      - "3000:3000"
+    depends_on:
+      - postgres
+  postgres:
+    image: postgres:11-alpine
+    environment:
+      - POSTGRES_PASSWORD=39dkdk3f93kkd93k20dl201kd83
+    ports:
+      - '5432:5432'
+    volumes:
+      - my_dbdata:/var/lib/postgresql/data
+volumes:
+  bundle_cache:
+  my_dbdata:
+```
+
+And a corresponding environment variables file called `.docker-development-vars` filled with the following vars that will make your development life easier:
+
+```
+SECRET_KEY_BASE=4dff85262e4bb057a9f067efd9774febb8ebb8079ede34263bfe10642382841b29663f042ec8350c0be8f54a8804d125fb7966e962130acc084fd204f89f0a2ec600
+DATABASE_URL=postgres://postgres:39dkdk3f93kkd93k20dl201kd83@postgres:5432
+```
```

Comment:
I have a long-term goal that I'd like to start implementing here: eliminate `.docker-development-vars` all together.

I've been setting the development database URL as a defaults in the application. See:

https://github.com/brandnewbox/friends-cbcap-portal/blob/master/config/database.yml#L14

And the secret key base can now live inside of credentials.

I think this will help folks think about:

* development environments being completely self-sustaining
* how to manage secrets vs config

```suggestion
```